lib/copy/__tests__/copy-preserve-time.test.js

```diff
@@ -47,8 +47,35 @@ describeIf64('copy', () => {
           done()
         })
       })
+
+      describe('> with a read-only file', () => {
+        const src = path.join(SRC_FIXTURES_DIR, readOnlyFile)
+        const dest = path.join(TEST_DIR, readOnlyFile)
+
+        beforeEach(() => assertPermission(src, '444', {
+          tryToSet: true,
```

Comment:
Tried with `444` still no luck.

Reply:
Not saying it is the cause of failure for this test, just wondering why using `beforeEach` hook here?!

Reply:
Here I'm trying to set the permission of the `src` file to `444` because that's what we need to test, right? - that whether or not those permissions will get copied. 

The same thing is done in [copy-permissions.test](https://github.com/jprichardson/node-fs-extra/blob/master/lib/copy/__tests__/copy-permissions.test.js#L78).

Reply:
I think the problem might be due to parent directories of the two files being different - `src` is in fixtures dir and `dest` is in `os.tempdir`. I re-wrote the tests so that it'll first write the `src` file in `tempdir` and then copy it again in the `tempdir` (as `dest`) and now both files have same permissions. It's only when one of the files was in `fixtures` dir that the problem occurred. 

I'm not sure why this is, I've very little knowledge about *nix file permissions. Or maybe I'm doing something else wrong, I don't know.